### DISCUSSION TITLE
Giving thanks and support

### DISCUSSION THREAD

**Discussion by Koujibear76** [webrcade/webrcade#25] [General]
Hey there! I created a GitHub account just for this... First of all, THANK YOU for this amazing project. There are four retro-gaming geeks in this house and right now we have a slightly convoluted way of sharing a central emulation library. Recently PLEX Arcade seemed to be the answer to our emulation prayers but that turned out to be an under-developed dud. Even that doesn't compare to the simplicity, versatility and performance of WebRcade. I am so stoked for this and all that is coming.

On that note, is there any way non-coding, normal humans such as I can help support this project? Is there a donations link or a Patreon page we can subscribe to? I think I speak for many on here in saying that if we can help, in any way, to make sure WebRcade is further developed, we will gladly do it.

Thank again!
-- J.

**Comment by PaulLuigi** [comment]
Gigantic THANK YOU for this service!!!  It's truly awesome how well "it just works" for many platforms and on many platforms.

I saw a video for this service a few months ago on Retro Games Corps' YouTube channel and I thought it sounded cool at the time, but it was a few months before I got around to giving it a try.  I was hoping Webrcade would be able to play an English translation of a game that was never released in the US (Marvelous - Mouhitotsu no Takarajima) which is notoriously tricky to emulate, and I've failed on other devices.  Amazingly, it runs like a dream on my iPad, full sound and gamepad support right out of the gate.  

Never in a million years would I have imagined I'd be able to play a Japanese Super Nintendo game, translated by fans, on an iPad, through the browser, with a multi-platform gamepad, looking and sounding excellent, for virtually no effort.  Truly an incredible achievement!

**Comment by raz0red** [comment]
Thank you so much for the kinds words! I really appreciate it! 

Please let me know if there are any features you are looking forward to, or would like to see.

Thanks again.

**Reply by RivermanPaul** [reply]
I’ve added likes or comments in the appropriate pages, but just to put it all in one place:

- Search game library by letter (jump around alphabetically)
- Search for games by typing a name.  Both these search features would hugely improve searching speed in a complete library for a platform (1000+ games).
- Save states, ideally mapped to default hotkeys (select + L, select + R is typical). This has been a feature since the very first emulator I played in the late 1990s, so it’s something I really miss.  Old games were super hard but save states make them playable.  Rewind function would be a wonderful icing on the cake!  The tiny local memory this requires would be trivial for a modern device.
- Dropbox save games and save states.  It would be amazing to be able to switch devices.  I play on my iPad at home, but abroad it'd be great to pick up where I left off on my phone.  And later, when you add more performance intensive systems, I'll probably play those on my Mac since it will have the power to handle them.  I don't know of any retro game platform that offers cloud saves and save states, and Webrcade is perfectly suited to deliver it.  It would make for a compelling reason to use Webrcade over other emulation devices and services.
- N64 support.  I have a couple retro game consoles, but the best they can handle is SNES, so this would let me easily play early 3d games via my iOS devices or Mac.

I know this is a long wish list, but Webrcade is awesome and has so much potential.  I would love to see the day when Webrcade replaces local emulators for 90% of use cases, and I think these features would push it forward a long way toward that goal.

**Reply by raz0red** [reply]
Thanks a lot for the feedback, I really appreciate it!

I totally agree with all of your points, and they will definitely be added in the future. 

I am currently focused on N64 support and Dropbox game saves. I am hoping to get both of those into the next major release (there may be bug fix releases prior to the next major).

After that, better UI support for finding games is something that I would like to add, so that should cover your first two points (improved search and navigation).

Finally, save states is something that will ultimately be added. However, I can't add that until Dropboxx save support is available. Using local storage for save states isn't really feasible as they are too big.

Thanks again for the feedback! I can't promise a specific timeline, but please know all of the features will make their way into webrcade eventually.

**Reply by RivermanPaul** [reply]
Fantastic, I look forward to it!  Again thank you so much for your extremely hard work!

**Comment by Emreler1** [comment]
I also want to thank you very much for this amazing service. I wish to ask like my friends if there is a Patreon page or some thing that we can contribute to you?
